upup/pkg/fi/authenticate.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fi
+
+// Authenticator generates authentication credentials for requests.
+type Authenticator interface {
+	CreateToken(body []byte) (string, error)
+}
+
+// VerifyResult is the result of a successfully verified request.
+type VerifyResult struct {
+	// Instance is the hostname of the instance.
+	Instance string
```

Comment:
We could name this `NodeName`, as I think that's what we've actually verified - it's approved to use this node name but not others.

Reply:
Renamed field.